BookNBlock/src/app/ui/offer/offer.component.ts

```diff
@@ -1,15 +1,61 @@
 import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { Offer } from '../../data/offer';
 
 @Component({
   selector: 'app-offer',
   templateUrl: './offer.component.html',
-  styleUrls: ['./offer.component.css']
+  styleUrls: ['./offer.component.scss']
 })
 export class OfferComponent implements OnInit {
 
-  constructor() { }
+  offerForm: FormGroup;
+
+  constructor() {
+    this.createOfferForm();
+  }
 
   ngOnInit() {
   }
 
+  private createOfferForm() {
+    this.offerForm = new FormGroup({
+      name: new FormControl('', Validators.required),
+      title: new FormControl('', [Validators.required, Validators.minLength(10)]),
+      description: new FormControl('', Validators.required),
+      date: new FormControl('', Validators.required),
+      toDate: new FormControl('', Validators.required),
+      prize: new FormControl('', Validators.required),
+      street: new FormControl('', Validators.required),
+      number: new FormControl('', Validators.required),
+      city: new FormControl('', Validators.required),
+      zip: new FormControl('', Validators.required),
+    });
+  }
+
+  onSubmit() {
+    console.log(JSON.stringify(this.offerForm.value));
+  }
+
+  prepareOffer() {
+    const formModel = this.offerForm.value;
```

Comment:
Der Transaction Service erwartet nur die Daten und baut dann das Offer auf. Er kennt auch die walletId. Die Id muss eg im SmartContract bestimmt werden.

Reply:
Es gibt bisher noch keine Funktionalität hierfür im Transaction Service

Reply:
hab gerade gesehen, dass das in deinem PR mit drin ist. Wenn deiner gemerged ist update ich diesen branch nochmal